spec/python-spec.coffee

```diff
@@ -296,3 +296,16 @@ describe "Python grammar", ->
       expect(tokens[1][0]).toEqual value: 'SELECT bar', scopes: ['source.python', scope]
       expect(tokens[2][0]).toEqual value: 'FROM foo', scopes: ['source.python', scope]
       expect(tokens[3][0]).toEqual value: delim, scopes: ['source.python', scope, 'punctuation.definition.string.end.python']
+  it "tokenizes a function definition with annotations", ->
+    tokens = grammar.tokenizeLines('def f(a: int) -> int:')
+
+    expect(tokens[0][0]).toEqual value: 'def', scopes: ['source.python', 'meta.function.python', 'storage.type.function.python']
+    expect(tokens[0][2]).toEqual value: 'f', scopes: ['source.python', 'meta.function.python', 'entity.name.function.python', 'support.function.magic.python']
+    expect(tokens[0][3]).toEqual value: '(', scopes: ['source.python', 'meta.function.python', 'punctuation.definition.parameters.begin.python']
+    expect(tokens[1][1]).toEqual value: 'a', scopes: ['source.python', 'meta.function.python', 'meta.function.parameters.python', 'variable.parameter.function.python']
+    expect(tokens[1][2]).toEqual value: ':', scopes: ['source.python', 'meta.function.python', 'meta.function.parameters.python', 'punctuation.definition.annotation.parameter.python']
```

Comment:
True, but this is not a dictionary... this is more like a `valuetriple`:

    def f(a:int=1): pass

`a` is the key, and both `1` and `int` are the associate values.

For tokenizing purposes, this might be splitting hairs... I'm more concerned with the fact that information about being "inside a definition" might be lost.

If `punctuation.separator.valuepair` would lead to better syntax highlighting I'd surely pick that